src/box/lua/config/applier/credentials.lua

```diff
@@ -57,15 +345,13 @@ local function create_roles(role_map)
     for role_name, role_def in pairs(role_map or {}) do
         if role_def ~= nil then
             create_role(role_name)
-            grant_privileges(role_name, role_def.privileges, 'role',
-                box.schema.role.grant)
         end
     end
 
     -- Assign underlying roles.
     for role_name, role_def in pairs(role_map or {}) do
         if role_def ~= nil then
-            assign_roles_to_role(role_name, role_def.roles)
+            sync_privileges(role_name, role_def, 'role')
         end
     end
 end
```

Comment:
The comment is outdated here and in the block just above.

Reply:
Fixed.